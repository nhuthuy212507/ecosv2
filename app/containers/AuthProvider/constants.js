/*
 *
 * AuthProvider constants
 *
 */

export const GET_PROFILE = 'app/UserProvider/GET_PROFILE';
export const GET_PROFILE_SUCCESS = 'app/UserProvider/GET_PROFILE_SUCCESS';
export const GET_PROFILE_ERROR = 'app/UserProvider/GET_PROFILE_ERROR';
