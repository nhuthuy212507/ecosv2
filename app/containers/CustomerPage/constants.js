/*
 *
 * CustomerPage constants
 *
 */

export const KEY = 'customerPage';

export const GET_LIST_CUSTOMER = 'app/CustomerPage/GET_LIST_CUSTOMER';
export const GET_LIST_CUSTOMER_SUCCESS =
  'app/CustomerPage/GET_LIST_CUSTOMER_SUCCESS';

export const GET_LIST_CUSTOMER_ERROR =
  'app/CustomerPage/GET_LIST_CUSTOMER_ERROR';

export const GET_OPTIONS_CUSTOMER_REFERENCE =
  'app/CustomerPage/GET_OPTIONS_CUSTOMER_REFERENCE';

export const GET_OPTIONS_CUSTOMER_REFERENCE_SUCCESS =
  'app/CustomerPage/GET_OPTIONS_CUSTOMER_REFERENCE_SUCCESS';

export const GET_OPTIONS_CUSTOMER_REFERENCE_ERROR =
  'app/CustomerPage/GET_OPTIONS_CUSTOMER_REFERENCE_ERROR';

export const GET_OPTIONS_CREATED_BY = 'app/CustomerPage/GET_OPTIONS_CREATED_BY';

export const GET_OPTIONS_CREATED_BY_SUCCESS =
  'app/CustomerPage/GET_OPTIONS_CREATED_BY_SUCCESS';

export const GET_OPTIONS_CREATED_BY_ERROR =
  'app/CustomerPage/GET_OPTIONS_CREATED_BY_ERROR';

export const ADD_CUSTOMER = 'app/CustomerPage/ADD_CUSTOMER';
export const ADD_CUSTOMER_SUCCESS = 'app/CustomerPage/ADD_CUSTOMER_SUCCESS';
export const ADD_CUSTOMER_ERROR = 'app/CustomerPage/ADD_CUSTOMER_ERROR';

export const GET_CUSTOMER_EDITTING = 'app/CustomerPage/GET_CUSTOMER_EDITTING';
export const GET_CUSTOMER_EDITTING_SUCCESS =
  'app/CustomerPage/GET_CUSTOMER_EDITTING_SUCCESS';
export const GET_CUSTOMER_EDITTING_ERROR =
  'app/CustomerPage/GET_CUSTOMER_EDITTING_ERROR';

export const EDIT_CUSTOMER = 'app/CustomerPage/EDIT_CUSTOMER';
export const EDIT_CUSTOMER_SUCCESS = 'app/CustomerPage/EDIT_CUSTOMER_SUCCESS';
export const EDIT_CUSTOMER_ERROR = 'app/CustomerPage/EDIT_CUSTOMER_ERROR';

export const GET_OPTIONS_CITY_WAREHOUSE_ADDRESS =
  'app/CustomerPage/GET_OPTIONS_CITY_WAREHOUSE_ADDRESS';

export const GET_OPTIONS_CITY_WAREHOUSE_ADDRESS_SUCCESS =
  'app/CustomerPage/GET_OPTIONS_CITY_WAREHOUSE_ADDRESS_SUCCESS';

export const GET_OPTIONS_CITY_WAREHOUSE_ADDRESS_ERROR =
  'app/CustomerPage/GET_OPTIONS_CITY_WAREHOUSE_ADDRESS_ERROR';

export const GET_OPTIONS_DISTRICT_WAREHOUSE_ADDRESS =
  'app/CustomerPage/GET_OPTIONS_DISTRICT_WAREHOUSE_ADDRESS';

export const GET_OPTIONS_DISTRICT_WAREHOUSE_ADDRESS_SUCCESS =
  'app/CustomerPage/GET_OPTIONS_DISTRICT_WAREHOUSE_ADDRESS_SUCCESS';

export const GET_OPTIONS_DISTRICT_WAREHOUSE_ADDRESS_ERROR =
  'app/CustomerPage/GET_OPTIONS_DISTRICT_WAREHOUSE_ADDRESS_ERROR';

export const GET_OPTIONS_WARD_WAREHOUSE_ADDRESS =
  'app/CustomerPage/GET_OPTIONS_WARD_WAREHOUSE_ADDRESS';

export const GET_OPTIONS_WARD_WAREHOUSE_ADDRESS_SUCCESS =
  'app/CustomerPage/GET_OPTIONS_WARD_WAREHOUSE_ADDRESS_SUCCESS';

export const GET_OPTIONS_WARD_WAREHOUSE_ADDRESS_ERROR =
  'app/CustomerPage/GET_OPTIONS_WARD_WAREHOUSE_ADDRESS_ERROR';

export const ADD_WAREHOUSE_ADDRESS = 'app/CustomerPage/ADD_WAREHOUSE_ADDRESS';
export const ADD_WAREHOUSE_ADDRESS_SUCCESS =
  'app/CustomerPage/ADD_WAREHOUSE_ADDRESS_SUCCESS';
export const ADD_WAREHOUSE_ADDRESS_ERROR =
  'app/CustomerPage/ADD_WAREHOUSE_ADDRESS_ERROR';

export const GET_LIST_WAREHOUSE_ADDRESS =
  'app/CustomerPage/GET_LIST_WAREHOUSE_ADDRESS';
export const GET_LIST_WAREHOUSE_ADDRESS_SUCCESS =
  'app/CustomerPage/GET_LIST_WAREHOUSE_ADDRESS_SUCCESS';
export const GET_LIST_WAREHOUSE_ADDRESS_ERROR =
  'app/CustomerPage/GET_LIST_WAREHOUSE_ADDRESS_ERROR';

export const EDIT_WAREHOUSE_ADDRESS = 'app/CustomerPage/EDIT_WAREHOUSE_ADDRESS';
export const EDIT_WAREHOUSE_ADDRESS_SUCCESS =
  'app/CustomerPage/EDIT_WAREHOUSE_ADDRESS_SUCCESS';
export const EDIT_WAREHOUSE_ADDRESS_ERROR =
  'app/CustomerPage/EDIT_WAREHOUSE_ADDRESS_ERROR';

export const DELETE_WAREHOUSE_ADDRESS =
  'app/CustomerPage/DELETE_WAREHOUSE_ADDRESS';
export const DELETE_WAREHOUSE_ADDRESS_SUCCESS =
  'app/CustomerPage/DELETE_WAREHOUSE_ADDRESS_SUCCESS';
export const DELETE_WAREHOUSE_ADDRESS_ERROR =
  'app/CustomerPage/DELETE_WAREHOUSE_ADDRESS_ERROR';
