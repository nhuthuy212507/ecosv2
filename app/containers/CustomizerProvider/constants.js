/*
 *
 * CustomizerProvider constants
 *
 */

export const TOGGLE_THEME = 'app/CustomizerProvider/TOGGLE_THEME';
