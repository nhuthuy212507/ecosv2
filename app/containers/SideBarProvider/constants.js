/*
 *
 * SideBarProvider constants
 *
 */

export const CHANGE_DESKTOP_SIDEBAR_VISIBILITY =
  'app/SideBarProvider/CHANGE_DESKTOP_SIDEBAR_VISIBILITY';

export const CHANGE_MOBILE_SIDEBAR_VISIBILITY =
  'app/SideBarProvider/CHANGE_MOBILE_SIDEBAR_VISIBILITY';
