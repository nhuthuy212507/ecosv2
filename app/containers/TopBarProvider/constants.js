/*
 *
 * TopBarProvider constants
 *
 */

export const DEFAULT_ACTION = 'app/TopBarProvider/DEFAULT_ACTION';
