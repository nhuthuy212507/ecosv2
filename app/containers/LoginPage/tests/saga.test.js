/**
 * Tests for LoginPage sagas
 */

import { takeLatest } from 'redux-saga/effects';

import { LOGIN } from '../constants';
import loginPageSaga, { login } from '../saga';

const user = {
  username: 'test',
  password: 'test',
};

/* eslint-disable redux-saga/yield-effects */
describe('login Saga', () => {
  let loginGenerator;

  beforeEach(() => {
    loginGenerator = login({ user });

    const selectDescriptor = loginGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = loginGenerator.next(user).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  describe('loginPage Saga', () => {
    const loginSaga = loginPageSaga();

    it('should start task to watch for LOAD_REPOS action', () => {
      const takeLatestDescriptor = loginSaga.next().value;
      expect(takeLatestDescriptor).toEqual(takeLatest(LOGIN, login));
    });
  });
});
