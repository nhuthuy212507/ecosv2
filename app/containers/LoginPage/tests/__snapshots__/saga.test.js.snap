// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login Saga loginPage Saga should start task to watch for LOAD_REPOS action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      Object {
        "data": Object {
          "password": "test",
          "username": "test",
        },
        "method": "post",
        "url": "/auth",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`login Saga loginPage Saga should start task to watch for LOAD_REPOS action 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "error": [TypeError: Cannot read property 'token' of undefined],
      "type": "app/LoginPage/LOGIN_ERROR",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;
